k4GaudiPandora/CMakeLists.txt

```diff
@@ -1,61 +1,40 @@
-#[[
-Copyright (c) 2020-2024 Key4hep-Project.
-
-This file is part of Key4hep.
-See https://key4hep.github.io/key4hep-doc/ for further info.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-]]
-
-set(_plugin_sources ${CMAKE_CURRENT_LIST_DIR}/
-					src/CalorimeterHitType.cc
-                    src/DDCaloDigi.cc
-                    src/DDScintillatorPpdDigi.cc
-					src/DDSimpleMuonDigi.cc
-					)
-
+#---------------------------------------------------------------
+FILE( GLOB _plugin_sources src/*.c* )
```

Comment:
I would be in favor of listing the source files explicitly. `GLOB` is an anti-pattern in CMake and probably doesn't do what you want in certain cases. From the [documentation](https://cmake.org/cmake/help/v3.7/command/file.html)

> Note We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate.

Reply:
Also not using glob let's you chose which files to compile, so the ones that don't compile you don't have to remove.